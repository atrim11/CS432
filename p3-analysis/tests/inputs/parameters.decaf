// Test Case: Function Definitions

// 1. Function with parameters not used in the body
def int unusedParameters(int a, int b) {
    return a + b;
}

def int longer_params(int a, void b, bool c) {
    return a + b + c;
}

def int main() {
    longer_params(2, NULL, 50, true);
    longer_params(1, 2);
    longer_params();
    b = longer_params;
result = multiply(undeclaredVar1, undeclaredVar2);  // Error: 'undeclaredVar1' and 'undeclaredVar2' are undeclared
sum = sumArray(unknownArray, unknownSize);  // Error: 'unknownArray' and 'unknownSize' are undeclared
final = multiply(multiply(5, 2), undeclaredFunction(1, 2));  // Error: 'undeclaredFunction' is undeclared
result = multiply(10);  // Error: multiply expects 2 parameters, got 1
return -1;
 x = 10;  // x is declared
 x = 20;  // Error: x is already declared
 undeclaredVar = x + undeclaredVar2;  // Error: 'undeclaredVar2' is undeclared
 totalSum = sumArray(arr, undeclaredSize);  // Error: 'undeclaredSize' is undeclared
 multiplyResult = multiply(3, undeclaredFloat);  // Error: 'undeclaredFloat' is undeclared
 multiplyWithFuncResult = multiply(3, someFunction());  // Error: 'someFunction' is undeclared
 messageResult = prMessage("Hello");  // Error: Cannot assign void function result to 
 nonExistent = nonExistentFunction();  // Error: 'nonExistentFunction' is undeclared
funcWithBadParam(1, 2);  // Error: Function defined with undeclared parameter types
 recResult = recursiveFunction(5);  // Error: 'undeclaredParam' is undeclared
 accessBlockVar = blockVar;  // Error: 'blockVar' is undeclared outside the block
 tempResult = undeclaredVar1 + undeclaredVar2;  // Error: 'undeclaredVar1' and 'undeclaredVar2' are undeclared
declaredVar = undeclaredAssignment;  // Error: 'undeclaredAssignment' is undeclared
 resultFromUndeclared = undeclaredReturnTypeFunction();  // Error: Function has no declared return type
 faultyResult = faultyFunction(3, 5, 10);  // Error: Third parameter is undeclared
 arraySum = sumArray(arr, undeclaredSize);  // Error: 'undeclaredSize' is undeclared
}

// Test Case: Function Definitions with Undeclared Parameters

def void functionWithBadParam(void param) {
    print_int(param);
}

def void funcWithBadParam(int param1, bool param2) {
    return param1 + param2;
}
def int recursiveFunction(bool undeclaredParam) {
    return undeclaredParam + recursiveFunction(undeclaredParam - 1);
}
def bool undeclaredReturnTypeFunction() {
    return 42;
}
def int faultyFunction(int a, int b, int undeclaredParam) {
    return a + b + undeclaredParam;
}