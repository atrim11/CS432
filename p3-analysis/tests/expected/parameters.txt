Type mismatch: int expected but void found on line 9
Type mismatch: int expected but bool found on line 9
Symbol 'NULL' undefined on line 13
Invalid number of function arguments on line 13
Invalid number of function arguments on line 14
Invalid number of function arguments on line 15
Symbol 'b' undefined on line 16
Function 'longer_params' accessed as a variable on line 16
Symbol 'result' undefined on line 17
Symbol 'multiply' undefined on line 17
Symbol 'undeclaredVar1' undefined on line 17
Symbol 'undeclaredVar2' undefined on line 17
Symbol 'sum' undefined on line 18
Symbol 'sumArray' undefined on line 18
Symbol 'unknownArray' undefined on line 18
Symbol 'unknownSize' undefined on line 18
Symbol 'final' undefined on line 19
Symbol 'multiply' undefined on line 19
Symbol 'multiply' undefined on line 19
Symbol 'undeclaredFunction' undefined on line 19
Symbol 'result' undefined on line 20
Symbol 'multiply' undefined on line 20
Symbol 'x' undefined on line 22
Symbol 'x' undefined on line 23
Symbol 'undeclaredVar' undefined on line 24
Symbol 'x' undefined on line 24
Symbol 'undeclaredVar2' undefined on line 24
Symbol 'totalSum' undefined on line 25
Symbol 'sumArray' undefined on line 25
Symbol 'arr' undefined on line 25
Symbol 'undeclaredSize' undefined on line 25
Symbol 'multiplyResult' undefined on line 26
Symbol 'multiply' undefined on line 26
Symbol 'undeclaredFloat' undefined on line 26
Symbol 'multiplyWithFuncResult' undefined on line 27
Symbol 'multiply' undefined on line 27
Symbol 'someFunction' undefined on line 27
Symbol 'messageResult' undefined on line 28
Symbol 'prMessage' undefined on line 28
Symbol 'nonExistent' undefined on line 29
Symbol 'nonExistentFunction' undefined on line 29
Type mismatch in parameter 1 of call to 'funcWithBadParam': expected bool but found int on line 30
Symbol 'recResult' undefined on line 31
Type mismatch in parameter 0 of call to 'recursiveFunction': expected bool but found int on line 31
Symbol 'accessBlockVar' undefined on line 32
Symbol 'blockVar' undefined on line 32
Symbol 'tempResult' undefined on line 33
Symbol 'undeclaredVar1' undefined on line 33
Symbol 'undeclaredVar2' undefined on line 33
Symbol 'declaredVar' undefined on line 34
Symbol 'undeclaredAssignment' undefined on line 34
Symbol 'resultFromUndeclared' undefined on line 35
Symbol 'faultyResult' undefined on line 36
Symbol 'arraySum' undefined on line 37
Symbol 'sumArray' undefined on line 37
Symbol 'arr' undefined on line 37
Symbol 'undeclaredSize' undefined on line 37
Type mismatch in parameter 0 of call to 'print_int': expected int but found void on line 43
Type mismatch: int expected but bool found on line 47
Invalid non-void return from void function on line 47
Type mismatch: int expected but bool found on line 50
Type mismatch in parameter 0 of call to 'recursiveFunction': expected bool but found int on line 50
Type mismatch: int expected but bool found on line 50
Type mismatch: bool expected but int found on line 53
